src/aiida/orm/implementation/storage_backend.py

```diff
@@ -304,6 +304,78 @@ def maintain(self, full: bool = False, dry_run: bool = False, **kwargs) -> None:
+
+    def _validate_or_init_backup_folder(self, dest, keep):
+        import json
+        import pathlib
+        import tempfile
+
+        from disk_objectstore import backup_utils
+
+        from aiida.common import exceptions
+        from aiida.storage.log import STORAGE_LOGGER
+
+        backup_info_fname = 'aiida-backup.json'
+        backup_info = {
+            'PROFILE_NAME': self.profile.name,
+            'PROFILE_UUID': self.profile.uuid,
+            'STORAGE_BACKEND': self.profile.storage_backend,
+        }
+
+        try:
+            # this creates the dest folder if it doesn't exist
+            backup_manager = backup_utils.BackupManager(dest, keep=keep)
+
+            backup_info_path = backup_manager.path / backup_info_fname
+            if backup_manager.check_path_exists(backup_info_path):
+                success, stdout = backup_manager.run_cmd(['cat', str(backup_info_path)])
+                if not success:
+                    raise exceptions.StorageBackupError(f"Couldn't read {backup_info_path!s}.")
+                backup_info_existing = json.loads(stdout)
+                if backup_info_existing != backup_info:
+                    raise exceptions.StorageBackupError(
+                        'The chosen destination contains backups of a different profile! Aborting!'
+                    )
+            else:
+                STORAGE_LOGGER.warning('Initializing a new backup folder.')
+                # make sure the folder is empty
+                success, stdout = backup_manager.run_cmd(['ls', '-A', str(backup_manager.path)])
+                if not success:
+                    raise exceptions.StorageBackupError(f"Couldn't read {backup_info_path!s}.")
+                if stdout:
+                    raise exceptions.StorageBackupError("Can't initialize the backup folder, destination is not empty.")
+
+                with tempfile.TemporaryDirectory() as tmpdir:
+                    tmp_path = pathlib.Path(tmpdir) / backup_info_fname
+                    with open(tmp_path, 'w', encoding='utf-8') as fhandle:
+                        json.dump(backup_info, fhandle, indent=4)
+                        fhandle.write('\n')
+                    backup_manager.call_rsync(tmp_path, backup_info_path)
+
+        except backup_utils.BackupError as exc:
+            raise exceptions.StorageBackupError(*exc.args) from exc
+
+    def backup(
+        self,
+        dest: str,
+        keep: Optional[int] = None,
+    ):
+        """Create a backup of the storage contents.
+
+        :param dest: The path to the destination folder.
+        :param keep: The number of backups to keep in the target destination, if the backend supports it.
+        :raises ValueError: If the input parameters are invalid.
+        :raises StorageBackupError: If an error occurred during the backup procedure.
+        """
+        self._validate_or_init_backup_folder(dest, keep)
+        self._backup_backend(dest, keep)
```

Comment:
One downside of doing it in this order is that know if the storage backend doesn't implement the backup and raises `NotImplementedError`, the backup folder are created anyway, including the transfer of the `aiida-backup.json` file. It would be great if we can check if the method is implemented first, but not sure how to go about it. Ideas?

Reply:
I see what you mean. One possibility is to checkthe file `aiida-backup.json` first, then backup, and then create the file. In the case of unimplemented backup, the destination folder will be created but no `aiida-backup.json`. However, if a backup somehow gets interrupted, then the `aiida-backup.json` would be missing as well, which might be useful (to know what profile backup was it & how to potentially resume).

2nd possibility is to keep track of whether  `aiida-backup.json`  was created, and in the case of NotImplementedError, delete it.

Another option is to have an additional property/method. i.e. something like `is_backup_implemented` that is `False` by default, and if backend implement a backup methods, they should overwrite this to `True`.

What do you think? (but also perhaps it's  fine if the folder & file are created even when notimplementederror is raised?)

Reply:
> as it's very simple to add and solves the issue.

Yes, but the plugin developer needs to not forget about this or it won't work. If they just implement the backup method but forget to override the `is_backup_implemented` property, things won't work as expected. Seems easy to overlook.

But okay, let's go with that solution for now.